### ISSUE TITLE
Fixed n_gram_range still not being used? 

### ISSUE THREAD

**Issue by jburdo1** [MaartenGr/BERTopic#1478]
Hi @MaartenGr , is it the case that the fixed 'n_gram_range' parameter that seems to have been deactivated in Version 0.2.2 is still not useable? Though it's still described in the 'Hyperparameter Tuning' section, it's not working for me, and I see that it was at one point removed from BERTopic. 

Cheers,
 
Joe

**Comment by jburdo1** [comment]
Sure thing.  That was the last mention of 'n_gram_search' I came across when doing a keyphrase search. 

`!pip install sentence_transformers
from sentence_transformers import SentenceTransformer

# Prepare embeddings
sentence_model = SentenceTransformer("all-MiniLM-L6-v2")
embeddings = sentence_model.encode(abstract_list, show_progress_bar=True)`

`from hdbscan import HDBSCAN
from umap import UMAP

umap_model = UMAP(n_neighbors=5, n_components=5, metric='cosine', random_state=72)
hdbscan_model = HDBSCAN(min_cluster_size=5, metric='euclidean', prediction_data=True)`

`#Declare addtional stop words to avoid topic formation
from sklearn.feature_extraction import text
from sklearn.feature_extraction.text import CountVectorizer
from bertopic.vectorizers import ClassTfidfTransformer

vectorizer_model = CountVectorizer(stop_words="english")
#my_additional_stop_words = ['the', 'to', 'is', 'ow', 'redacted', 'tis']
#stop_words = text.ENGLISH_STOP_WORDS.union(my_additional_stop_words)
#vectorizer_model = CountVectorizer(stop_words=list(stop_words))
ctfidf_model = ClassTfidfTransformer(reduce_frequent_words=True)

topic_model = BERTopic(n_gram_range = (2,3), min_topic_size = 5, ctfidf_model = ctfidf_model, nr_topics = 'None', umap_model = umap_model, hdbscan_model = hdbscan_model, vectorizer_model=vectorizer_model, calculate_probabilities=True, verbose=True)
topics, probs = topic_model.fit_transform(abstract_list, embeddings)`

<img width="255" alt="Screenshot 2023-08-21 142458" src="https://github.com/MaartenGr/BERTopic/assets/97640624/aee17adb-ed9e-4b2d-b974-37cb2b80715e">

Setting n_gram_range = (2,3) should force a minimum of bigrams, correct?

**Comment by MaartenGr** [comment]
> Setting n_gram_range = (2,3) should force a minimum of bigrams, correct?

The `n_gram_range` parameter only works if you did not use a `CountVectorizer` which overwrites that parameter. You should use the `n_gram_range` parameter in the CountVectorizer instead.

**Comment by jburdo1** [comment]
That works, thanks! 

In case someone else comes across this issue in the future, the CountVectorizer parameter is 'ngram_range' instead of 'n_gram_range'
